### PR TITLE
Fix labels used by Prometheus Operator to load

### PR DESCRIPTION
**Pull request by nlamirault**
- used additional labels for ServiceMonitor and PrometheusRule used by the Prometheus Operator
- typo on PrometheusRule field

### CODE REVIEW

File: charts/quickwit/templates/prometheusrule.yaml [hunk 1]
```diff
@@ -5,9 +5,12 @@ metadata:
   name: {{ include "quickwit.fullname" . }}
   labels:
     {{- include "quickwit.labels" . | nindent 4 }}
+    {{- with .Values.prometheusRule.additionalLabels }}
```
Comment (rdettai) [inline charts/quickwit/templates/prometheusrule.yaml:8]: can we add `prometheusRule.additionalLabels` to the `values.yaml` ?

File: charts/quickwit/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -5,6 +5,9 @@ metadata:
   name: {{ include "quickwit.fullname" . }}
   labels:
     {{- include "quickwit.labels" . | nindent 4 }}
+    {{- with .Values.serviceMonitor.additionalLabels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
```
Comment (rdettai) [inline charts/quickwit/templates/servicemonitor.yaml:10]: do we really need an extra set of additionalLabels? Can't the Prometheus operator be configured to use the `quickwit.labels` which are already configurable?

File: charts/quickwit/templates/prometheusrule.yaml (hunk 1 above)
Comment (nlamirault) [reply charts/quickwit/templates/prometheusrule.yaml:8]: oh i forget that. done.

File: charts/quickwit/templates/servicemonitor.yaml (hunk 2 above)
Comment (nlamirault) [reply charts/quickwit/templates/servicemonitor.yaml:10]: It's a common pattern in the Prometheus Operator community to have these labels for ServiceMonitor, PodMonitor, PrometheusRule, ...
Ex:
- https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-elasticsearch-exporter/templates/prometheusrule.yaml
- https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-elasticsearch-exporter/templates/podmonitor.yaml
- https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-elasticsearch-exporter/templates/servicemonitor.yaml

**Review by rdettai — APPROVED** [review]
Merging this as is, but opened https://github.com/quickwit-oss/helm-charts/issues/82 to track follow up work.

**Pull request merged**